### DISCUSSION TITLE
agent status via slack status

### DISCUSSION THREAD

**Discussion by psifertex** [chatwoot/chatwoot#8446] [Feature Requests]
### Is your feature or enhancement related to a problem? Please describe.

Right now, I have to be logged into the web interface to set my status.

### Describe the solution you'd like

I'd like status to be sync'ed from the slack API.

### Describe alternatives you've considered

Manually logging back in is a pain.

### Additional context

This issue was previously tracked at https://github.com/chatwoot/chatwoot/issues/8184 but was closed due to no progress. Still a requested feature

**Comment by pranavrajs** [comment]
@psifertex I don't think this is supported in the Events API. https://api.slack.com/docs/presence-and-status#user-presence-and-status__determining-user-presence__presence-subscriptions-over-events-api

**Reply by psifertex** [reply]
I's not available via webhooks. It absolutely is supported over the Events API. 

See my previous reply which quotes from their help page:

> RTM API Presence is now only available via subscription. As of January 2018, [presence_change](https://api.slack.com/events/presence_change) events are not dispatched without [presence subscriptions](https://api.slack.com/docs/presence-and-status) established with [presence_sub](https://api.slack.com/events/presence_sub). Relatedly, current user presence status is no longer communicated in [rtm.start](https://api.slack.com/methods/rtm.start). [Learn more](https://api.slack.com/changelog/2018-01-prese).

**Reply by psifertex** [reply]
In other words, you set up a presence subscription event and you get notifications whenever it changes.

**Reply by pranavrajs** [reply]
Maybe I'm getting this wrong, but do they send us a webhook update when someone in the channel changes the status?

**Reply by psifertex** [reply]
Yes, the data comes back as a webhook. However it's just not the simple webhook setup older plugins can use. You have to register using https://api.slack.com/events/presence_sub first.

Here's a ruby library that implements this type of plugin as an example: 

https://github.com/slack-ruby/slack-ruby-bot-server-rtm
